### ISSUE TITLE
SE3.translation() has incorrect backward for pp.se3(...).Exp() in v0.9.0

### ISSUE THREAD

**Issue by MarkChenYutian** [pypose/pypose#395]
### 🐛 Describe the bug

In PyPose v0.9.0, `SE3.translation()` appears to return the correct forward value but an incorrect backward pass when the SE(3) element is produced from a tangent parameter via `pp.se3(...).Exp()`.

More concretely, for `xi ∈ se3`, the path

```python
xi.Exp().translation()
```

produces a Jacobian w.r.t. xi whose last three columns (the rotational tangent part) do not match finite differences.

This seems specific to the .translation() accessor. An equivalent forward computation using

```python
xi.Exp().Act(torch.zeros(3))
```

returns the same translation value, but its Jacobian matches finite differences.

From source inspection, this looks related to SE3Type.translation() returning a raw slice of the internal storage, while other SE(3) ops go through custom autograd paths.

Minimal reproducer:

```python
import torch
import pypose as pp
from pypose.optim.functional import modjac

torch.manual_seed(42)
dtype = torch.float64

class Probe(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.xi = pp.Parameter(
            pp.se3(torch.tensor([0.2, 0.3, 0.1, 0.2, 0.3, 0.1], dtype=dtype))
        )

    def forward(self):
        return self.xi.Exp().translation()

def finite_difference_jacobian(module, eps=1e-7):
    p = module.xi
    J = torch.zeros(3, 6, dtype=dtype)
    for i in range(6):
        p.data.view(-1)[i] += eps
        rp = module().clone()

        p.data.view(-1)[i] -= 2 * eps
        rm = module().clone()

        p.data.view(-1)[i] += eps
        J[:, i] = (rp - rm) / (2 * eps)
    return J

m = Probe()

J_auto = modjac(m, input=(), flatten=False, vectorize=True)
J_auto = (J_auto[0] if isinstance(J_auto, tuple) else J_auto).reshape(3, 6)

J_fd = finite_difference_jacobian(Probe())

print("FD:\n", J_fd)
print("\nautograd:\n", J_auto)
print("\nmax abs diff =", (J_auto - J_fd).abs().max().item())
```

Observed result on v0.9.0:
```
FD:
 tensor([[ 0.9834, -0.0395,  0.1516,  0.0166,  0.0395, -0.1516],
         [ 0.0593,  0.9917, -0.0939, -0.0593,  0.0083,  0.0939],
         [-0.1449,  0.1038,  0.9785,  0.1449, -0.1038,  0.0215]],
       dtype=torch.float64)

autograd:
 tensor([[ 0.9834, -0.0395,  0.1516, -0.0329, -0.0285,  0.1514],
         [ 0.0593,  0.9917, -0.0939,  0.0680, -0.0164, -0.0867],
         [-0.1449,  0.1038,  0.9785, -0.1382,  0.1064, -0.0427]],
       dtype=torch.float64)

max abs diff = 0.3029
```

What I observe:

The first three columns (d translation / d rho) match finite differences.
The last three columns (d translation / d phi) do not match finite differences.
The forward value itself is correct; the issue is in the backward pass.

### Versions

```
PyTorch version: 2.8.0+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.12.9 | packaged by Anaconda, Inc. | (main, Feb  6 2025, 18:56:27) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-6.17.0-20-generic-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 5090
Nvidia driver version: 580.126.09
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==2.4.0
[pip3] torch==2.8.0
[pip3] torchvision==0.23.0
[conda] numpy                     2.4.0                    pypi_0    pypi
[conda] torch                     2.8.0                    pypi_0    pypi
[conda] torchvision               0.23.0                   pypi_0    pypi
```

**Comment by MarkChenYutian** [comment]
Verification of problem:

```python
import torch
import pypose as pp
from pypose.optim.functional import modjac

torch.manual_seed(42)
dtype = torch.float64

class Probe(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.xi = pp.Parameter(
            pp.se3(torch.tensor([0.2, 0.3, 0.1, 0.2, 0.3, 0.1], dtype=dtype))
        )

    def forward(self):
        return self.xi.Exp().translation()
        # return self.xi.Exp().Act(torch.zeros(3, dtype=dtype))

def finite_difference_jacobian(module, eps=1e-7):
    p = module.xi
    J = torch.zeros(3, 6, dtype=dtype)
    for i in range(6):
        p.data.view(-1)[i] += eps
        rp = module().clone()

        p.data.view(-1)[i] -= 2 * eps
        rm = module().clone()

        p.data.view(-1)[i] += eps
        J[:, i] = (rp - rm) / (2 * eps)
    return J

m = Probe()

J_auto = modjac(m, input=(), flatten=False, vectorize=True)
J_auto = (J_auto[0] if isinstance(J_auto, tuple) else J_auto).reshape(3, 6)

J_fd = finite_difference_jacobian(Probe())

print("FD:\n", J_fd)
print("\nautograd:\n", J_auto)
print("\nmax abs diff =", (J_auto - J_fd).abs().max().item())
```

Have result:

```
FD:
 tensor([[ 0.9834, -0.0395,  0.1516,  0.0166,  0.0395, -0.1516],
        [ 0.0593,  0.9917, -0.0939, -0.0593,  0.0083,  0.0939],
        [-0.1449,  0.1038,  0.9785,  0.1449, -0.1038,  0.0215]],
       dtype=torch.float64, grad_fn=<CopySlices>)

autograd:
 tensor([[ 0.9834, -0.0395,  0.1516, -0.0329, -0.0285,  0.1514],
        [ 0.0593,  0.9917, -0.0939,  0.0680, -0.0164, -0.0867],
        [-0.1449,  0.1038,  0.9785, -0.1382,  0.1064, -0.0427]],
       dtype=torch.float64)

max abs diff = 0.30293271343808564
```

While replace the `forward()` impl with `return self.xi.Exp().Act(torch.zeros(3, dtype=dtype))` gives the result

```
FD:
 tensor([[ 0.9834, -0.0395,  0.1516,  0.0166,  0.0395, -0.1516],
        [ 0.0593,  0.9917, -0.0939, -0.0593,  0.0083,  0.0939],
        [-0.1449,  0.1038,  0.9785,  0.1449, -0.1038,  0.0215]],
       dtype=torch.float64, grad_fn=<CopySlices>)

autograd:
 tensor([[ 0.9834, -0.0395,  0.1516,  0.0166,  0.0395, -0.1516],
        [ 0.0593,  0.9917, -0.0939, -0.0593,  0.0083,  0.0939],
        [-0.1449,  0.1038,  0.9785,  0.1449, -0.1038,  0.0215]],
       dtype=torch.float64)

max abs diff = 2.0335586513320347e-10
```

**Comment by zitongzhan** [comment]
I will look into this. 0.9 indeed updated some gradient calculation of lietensor operations.

**Comment by zitongzhan** [comment]
I checked that this issue is not newly introduced in v0.9.0; it is already present in v0.7.x (at least 0.7.5).
